### PR TITLE
Allow docker as container-runtime up to k8s v1.24

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Allow docker as container-runtime up to k8s v1.24

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Allow docker as container-runtime up to k8s v1.24
```

### CODE REVIEW

File: pkg/tasks/probes.go [hunk 1]
```diff
@@ -90,7 +90,9 @@ func safeguard(s *state.State) error {
 
 		if nodesContainerRuntime != configuredClusterContainerRuntime {
 			errMsg := "Migration is not supported yet"
-			if cr.Containerd != nil {
+			if nodesContainerRuntime == "docker" {
+				errMsg = "Support for docker will be removed with Kubernetes 1.24 release. It is recommended to switch to containerd as container runtime using `kubeone migrate to-containerd`. To continue using Docker please specify ContainerRuntime explicitly in KubeOneCluster manifest"
```
Comment (ahmedwaleedmalik) [inline pkg/tasks/probes.go:94]: Let me know what you think of this @xmudrii @kron4eg, I'm divided on this one. This is the scenario where a user running docker is trying to upgrade to k8s v.1.22.x. If he hasn't explicitly specified the `containerRuntime` in cluster config then it's a little hard, still do-able maybe, to determine if `containerd` was a result of defaulting or was explicitly set here.

Comment (kron4eg) [reply pkg/tasks/probes.go:94]: sounds good

**Review by kron4eg — APPROVED** [review]
/approve

/hold
/assign @xmudrii to review

File: pkg/tasks/probes.go (hunk 1 above)
Comment (xmudrii) [reply pkg/tasks/probes.go:94]: Sounds good.

File: pkg/tasks/probes.go [hunk 2]
```diff
@@ -174,10 +176,10 @@ func runProbes(s *state.State) error {
 		return nil
 	}
 
-	gteKube121Condition, _ := semver.NewConstraint(">= 1.21")
+	gteKube124Condition, _ := semver.NewConstraint(">= 1.24")
```
Comment (xmudrii) [inline pkg/tasks/probes.go:179]: If I'm not mistaken, this is doing defaulting. As we discussed, we want to containerd by default for new 1.21+ clusters, so this shouldn't be changed.

I'll double check and get back to you.

**Review by xmudrii — CHANGES_REQUESTED** [review]

**Pull request merged**